service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/validator/SupervisionOrderEndDateValidator.java

```diff
@@ -0,0 +1,103 @@
+package uk.gov.hmcts.reform.fpl.service.orders.validator;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.enums.orders.SupervisionOrderEndDateType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.event.ManageOrdersEventData;
+import uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.time.LocalTime.MIDNIGHT;
+import static uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock.SUPERVISION_ORDER_END_DATE;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class SupervisionOrderEndDateValidator implements QuestionBlockOrderValidator {
+    private static final Integer MINIMUM_MONTHS_VALID = 1;
+    private static final Integer MAXIMUM_MONTHS_VALID = 12;
+    private static final String INVALID_TIME_MESSAGE = "Enter a valid time";
+    private static final String FUTURE_DATE_MESSAGE = "Enter an end date in the future";
+    private static final String END_DATE_MAX_RANGE_MESSAGE = "Supervision orders cannot last longer than 12 months";
+    private static final String END_DATE_MIN_RANGE_MESSAGE = "*******Supervision orders in months should be at least 1";
```

Comment:
did you mean to include the `*******` at the start of this message?

Reply:
Yes, it seems to be a styling markup in CCD

Reply:
this was initially put it as i beleived it was a formatting necessity but now that i know it is not, its been removed